src/openssl/ssl/server.cr

```diff
@@ -0,0 +1,42 @@
+require "socket"
+
+class OpenSSL::SSL::Server
+  include ::Socket::Server
+
+  getter wrapped : ::Socket::Server
+  getter context : OpenSSL::SSL::Context::Server
+
+  # If `#sync_close?` is `true`, closing this server will
+  # close the wrapped server.
+  property? sync_close : Bool
+
+  getter? closed : Bool = false
+
+  def initialize(@wrapped : ::Socket::Server, @context : OpenSSL::SSL::Context::Server = OpenSSL::SSL::Context::Server.new, @sync_close : Bool = true)
+  end
+
+  def self.open(wrapped : ::Socket::Server, context : OpenSSL::SSL::Context::Server = OpenSSL::SSL::Context::Server.new, sync_close : Bool = true)
+    server = new(wrapped, context, sync_close)
+
+    begin
+      yield server
+    ensure
+      server.close
+    end
+  end
+
+  def accept? : OpenSSL::SSL::Socket::Server?
```

Comment:
We need to provide `def accept` too. In `TCPServer` this comes from `Socket` (which we don't inherit).

`::Socket::Server` also needs to gain the correct `abstract def`s.